extensions/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx

```diff
@@ -4,44 +4,56 @@
 *--------------------------------------------------------------------------------------------*/
 
 import * as React from "react";
-import { AbstractWidgetProps, StagePanelLocation, StagePanelSection, StageUsage, UiItemsProvider } from "@itwin/appui-abstract";
-import { Localization } from "@itwin/core-common";
-import { ConfigurableCreateInfo, WidgetControl } from "@itwin/appui-react";
-import { IModelApp } from "@itwin/core-frontend";
-import { MapFeatureInfoOptions } from "./Interfaces";
+import { AbstractWidgetProps, AbstractZoneLocation, CommonToolbarItem, StagePanelLocation, StagePanelSection, StageUsage, ToolbarOrientation, ToolbarUsage, UiItemsProvider, WidgetState } from "@itwin/appui-abstract";
+import { ToolbarHelper, UiFramework } from "@itwin/appui-react";
 import { MapFeatureInfoWidget } from "./widget/FeatureInfoWidget";
+import { MapFeatureInfoOptions } from "./Interfaces";
+import { MapLayersUI } from "../mapLayers";
+import { DefaultMapFeatureInfoTool, getDefaultMapFeatureInfoToolItemDef } from "./MapFeatureInfoTool";
 
-export class  FeatureInfoUiItemsProvider implements UiItemsProvider {
+export class FeatureInfoUiItemsProvider implements UiItemsProvider {
   public readonly id = "FeatureInfoUiItemsProvider";
-  public static localization: Localization;
+  public static readonly widgetId = "map-layers:mapFeatureInfoWidget";
+
+  public constructor(private _featureInfoOpts?: MapFeatureInfoOptions) { }
```

Comment:
We might want to make `_featureInfoOpts ` mandatory since onMapHit should be mandatory.

Reply:
updated